backend/src/main/java/com/redhat/cloud/notifications/db/EndpointResources.java

```diff
@@ -126,24 +127,29 @@ public Endpoint getOrCreateEmailSubscriptionEndpoint(String accountId, EmailSubs
         return createEndpoint(endpoint);
     }
 
-    public Long getEndpointsCountPerCompositeType(String tenant, Set<CompositeEndpointType> type, Boolean activeOnly) {
-        return EndpointResources.queryBuilderEndpointsPerType(tenant, type, activeOnly)
+    public Long getEndpointsCountPerCompositeType(String tenant, @Nullable String name, Set<CompositeEndpointType> type, Boolean activeOnly) {
+        return EndpointResources.queryBuilderEndpointsPerType(tenant, name, type, activeOnly)
                 .buildCount(entityManager::createQuery)
                 .getSingleResult();
     }
 
-    public List<Endpoint> getEndpoints(String tenant, Query limiter) {
+    public QueryBuilder<Endpoint> getEndpointsQuery(String tenant, @Nullable String name) {
+        return QueryBuilder.builder(Endpoint.class)
+                .alias("e")
+                .where(
+                        WhereBuilder.builder()
+                                .and("e.accountId = :accountId", "accountId", tenant)
+                                .ifAnd(name != null && !name.isEmpty(), "LOWER(e.name) LIKE :name", "name", "%" + name.toLowerCase() + "%")
```

Comment:
`name.toLowerCase()` is evaluated before being passed to the `ifAnd` method and when `name` is `null`, then the evaluation throws a NPE.